src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1727,13 +1718,13 @@ private void LoadNodeModels(List<TypeLoadData> nodes, bool isPackageMember)
             }
         }
 
-        private IPreferences CreateOrLoadPreferences(IPreferences preferences)
+        public static IPreferences CreateOrLoadPreferences(IPreferences preferences, bool isServiceMode, string preferenceFilePath)
```

Comment:
Is this have to be public, or internal? Also comments please for any public functions/properties